pkg/scheduler/internal/queue/scheduling_queue_test.go

```diff
@@ -733,24 +724,25 @@ func Test_InFlightPods(t *testing.T) {
 				}
 				wantInFlightEvents = append(wantInFlightEvents, value)
 			}
-			if diff := cmp.Diff(wantInFlightEvents, listToValues(q.inFlightEvents), cmp.AllowUnexported(clusterEvent{})); diff != "" {
+			if diff := cmp.Diff(wantInFlightEvents, q.activeQ.listInFlightEvents(), cmp.AllowUnexported(clusterEvent{})); diff != "" {
 				t.Errorf("Unexpected diff in inFlightEvents (-want, +got):\n%s", diff)
 			}
 
 			if test.wantActiveQPodNames != nil {
-				podInfos := q.activeQ.List()
-				if len(podInfos) != len(test.wantActiveQPodNames) {
-					diff := cmp.Diff(test.wantActiveQPodNames, podInfos, cmpopts.SortSlices(func(a, b interface{}) bool {
-						return a.(framework.PodInfo).Pod.Name < b.(framework.PodInfo).Pod.Name
-					}))
-					t.Fatalf("Length of activeQ is not expected. Got %v, want %v.\n%s", len(podInfos), len(test.wantActiveQPodNames), diff)
+				pods := q.activeQ.list()
+				if len(pods) != len(test.wantActiveQPodNames) {
```

Comment:
odd that we only check the diff if the length doesn't match... we should always check the diff.

Reply:
Right, fixed to check the diff directly in two places

Reply:
The messages in `t.Fatalf` are no longer accurate.

Reply:
Fixed